### ISSUE TITLE
Compiler doesn't enforce init-only property assignments in a copy constructor

### ISSUE THREAD

**Issue by tmat** [dotnet/roslyn#68346]
An explicitly declared copy-constructor of a record type is not forced to initialize the properties defined in the primary constructor. It is thus possible to leave them unassigned and also defeat null checking analysis.

The following code should report an error:

```C#
#nullable enable

public record C(string X)
{
    protected C(C other)
    {
        Console.Write("copy ctor");
    }
}
```
